pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientCnx.java

```diff
@@ -252,6 +254,8 @@ public ClientCnx(ClientConfigurationData conf, EventLoopGroup eventLoopGroup, in
         this.state = State.None;
         this.protocolVersion = protocolVersion;
         this.idleState = new ClientCnxIdleState(this);
+        this.clientVersion = "Pulsar-Java-v" + PulsarVersion.getVersion()
+                + (conf.getDescription() == null ? "" : conf.getDescription());
```

Comment:
According to the PIP and the method description document:
> For example, if the client version is 3.0.0, and the description is "forked", the final client version string
>    * will be "Pulsar-Java-v3.0.0-forked".

We need a middle '-' here.